Update experimental feature note (#44457)
fleetdm/fleet: handbook/company/product-groups.md

1. Apply the `~experimental` label to all associated user stories.
2. Set the optional `isExperimental` property to "yes" in [pricing-features-table.yml](https://github.com/fleetdm/fleet/blob/main/handbook/company/pricing-features-table.yml).

> 3. Make sure all API endpoints and configuration surface documentation contain the following message (including the anticipated version it will be marked stable):

Reviewer: ```suggestion
3. Make sure all API endpoints and configuration surface documentation contains the following message (including the anticipated version it will be marked stable):
```

> > **Experimental feature**. This feature is undergoing rapid improvement, which may result in breaking changes to the API or configuration surface. It is not recommended for use in automated workflows. This feature's experimental satus will be reevaluated in Fleet 4.x.

Reviewer: ```suggestion
> **Experimental feature**. This feature is undergoing rapid improvement, which may result in breaking changes to the API or configuration surface. It is not recommended for use in automated workflows. This feature's experimental status will be reevaluated in Fleet 4.x.
```

> > **Experimental feature**. This feature is undergoing rapid improvement, which may result in breaking changes to the API or configuration surface. It is not recommended for use in automated workflows. This feature's experimental satus will be reevaluated in Fleet 4.x.

Reviewer 2: ```suggestion
> **Experimental feature**. This feature is undergoing rapid improvement, which may result in breaking changes to the API or configuration surface. It is not recommended for use in automated workflows. This feature's experimental status will be reevaluated in Fleet 4.x.
```

> > **Experimental feature**. This feature is undergoing rapid improvement, which may result in breaking changes to the API or configuration surface. It is not recommended for use in automated workflows. This feature's experimental status will be reevaluated in Fleet 4.x.

Reviewer: What does 4.x mean?

> > **Experimental feature**. This feature is undergoing rapid improvement, which may result in breaking changes to the API or configuration surface. It is not recommended for use in automated workflows. This feature's experimental status will be reevaluated in Fleet 4.x.

Author: ```suggestion
> **Experimental feature**. This feature is undergoing rapid improvement, which may result in breaking changes to the API or configuration surface. It is not recommended for use in automated workflows. This feature's experimental status will be reevaluated in Fleet 4.89.0.
```
